### ISSUE TITLE
`IfConditionHelper`: AssignOp\Coalesce shouldn't pops up "Do not compare objects directly" error

### ISSUE THREAD

**Issue by Slamdunk** [voku/phpstan-rules#26]
#### What is this feature about (expected vs actual behaviour)?

Expected: no issue.

Actual:

```
Coalesce: Do not compare objects directly, ACoalesceClass and ACoalesceInterface|null found.
Coalesce: Do not compare objects directly, ACoalesceInterface|null and ACoalesceClass found.
```

#### How can I reproduce it?

```php
interface ACoalesceInterface {}
class ACoalesceClass implements ACoalesceInterface {}

class AFoo {
    public function __construct(
        private ?ACoalesceInterface $var
    ) {
        $this->var ??= new ACoalesceClass();
    }
}
```

#### Does it take minutes, hours or days to fix?

I tried to reproduce it in a PR, but the error never pops up in the test suite, probably I can figure out the proper position for the test case.

#### Any additional information?

PHP: 8.1
`voku/phpstan-rules`: 3.1.1

**Comment by voku** [comment]
Hi, I tried to fix it in version 3.1.2 

... but I am sure there must be a better way with PHPStan to check if one type is an interface and the other type has implemented it. `accepts` is not working here because we want to check, e.g. "__toString" vs "string" types.  :thinking:

**Comment by voku** [comment]
fixed: "accepts" is working here, I only missed that it returns an `\PHPStan\TrinaryLogic` object and not `bool` :facepalm:
